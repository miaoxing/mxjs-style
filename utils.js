/**
 * NOTE: 仅支持小部分变量的转换，其他需要可以再加上
 */

function generateBootstrapVars(theme) {
  const vars = {
    blue: theme.colors.blue['500'],
    green: theme.colors.green['500'],
    yellow: theme.colors.yellow['500'],
    red: theme.colors.red['500'],
    cyan: theme.colors.cyan['500'],
  };

  let content = '';
  for (const name in vars) {
    content += `$${name}: ${vars[name]} !default;\n`;
  }
  return content;
}

module.exports = {
  generateBootstrapVars,
};
