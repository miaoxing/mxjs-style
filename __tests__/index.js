import {generateBootstrapVars} from '../utils';
import theme from '../theme-preset';

describe('style', () => {
  test('generateBootstrapVars', () => {
    const result = generateBootstrapVars(theme);
    expect(result).toMatchSnapshot();
  });
});

