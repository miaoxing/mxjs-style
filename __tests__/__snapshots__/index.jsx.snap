// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`style generateBootstrapVars 1`] = `
"$blue: #1cb7f0 !default;
$green: #5cb85c !default;
$yellow: #f28c48 !default;
$red: #fa5b50 !default;
$cyan: #5bc0de !default;
"
`;
